releases.go

```diff
@@ -0,0 +1,202 @@
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+)
+
+// ReleasesService handles communication with the releases methods
+// of the GitLab API.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/releases/index.html
+type ReleasesService struct {
+	client *Client
+}
+
+// Release represents a project release.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/releases/index.html#list-releases
+type Release struct {
+	TagName         string     `json:"tag_name"`
+	Name            string     `json:"name"`
+	Description     string     `json:"description,omitempty"`
+	DescriptionHTML string     `json:"description_html,omitempty"`
+	CreatedAt       *time.Time `json:"created_at,omitempty"`
+	Author          *struct {
+		ID        int    `json:"id"`
+		Name      string `json:"name"`
+		Username  string `json:"username"`
+		State     string `json:"state"`
+		AvatarURL string `json:"avatar_url"`
+		WebURL    string `json:"web_url"`
+	} `json:"author,omitempty"`
+	Commit *Commit `json:"commit,omitempty"`
+	Assets *struct {
```

Comment:
Is there a reason why you made this a pointer?

Reply:
I can't remember the exact thought process for this, probably I was just unsure if Assets will always be populated. But it looks like it will be. I guess pointer can be removed from Commit, too. And of course the author

Reply:
In all other structs we use `struct` instead of `*struct`, even if the field isn't populated. If the field isn't populated the struct would be initialized using it zero values, so it wouldn't break anything. This also prevents the need to check for `nil` pointers.

Reply:
Feel free to make a force push, as long as you don't change the code in that push (as I just reviewed and approved the PR 🎉)

Let me know when your done so I can merge the PR. Thanks!

Reply:
The rewrite is done.

Double checked that the diff to the original was empty before resetting releases-pr, so the code should be exactly the same.